src/Analysis/Ast/Impl/Specializations/Typing/Types/TypingTupleType.cs

```diff
@@ -28,16 +29,20 @@ internal class TypingTupleType : PythonCollectionType, ITypingTupleType {
         /// Creates type info for a strongly-typed tuple, such as Tuple[T1, T2, ...].
         /// </summary>
         /// <param name="itemTypes">Tuple item types.</param>
+        /// <param name="declaringModule">Declaring module. If null, then 'typing' is assumed.</param>
         /// <param name="interpreter">Python interpreter.</param>
-        public TypingTupleType(IReadOnlyList<IPythonType> itemTypes, IPythonInterpreter interpreter)
-            : base(null, BuiltinTypeId.Tuple, interpreter.ModuleResolution.GetSpecializedModule("typing"), false) {
-            ItemTypes = itemTypes;
+        public TypingTupleType(IReadOnlyList<IPythonType> itemTypes, IPythonModule declaringModule, IPythonInterpreter interpreter)
+            : base(BuiltinTypeId.Tuple, declaringModule ?? interpreter.ModuleResolution.GetSpecializedModule("typing"), false) {
+            ItemTypes = itemTypes.Count > 0 ? itemTypes : new[] { interpreter.UnknownType };
```

Comment:
Occasionally during restore we may get `Tuple[]`. This code works around this. I will continue looking into why thing might be happening.